ng-dev/pr/merge/strategies/autosquash-merge.ts

```diff
@@ -30,9 +30,8 @@ export class AutosquashMergeStrategy extends MergeStrategy {
    * @throws {PullRequestFailure} A pull request failure if the the pull request could not
    *   be merged and the pull request is misconfigured.
    */
-  override async merge(pullRequest: PullRequest): Promise<void> {
-    const {prNumber, targetBranches, requiredBaseSha, needsCommitMessageFixup, githubTargetBranch} =
-      pullRequest;
+  override async check(pullRequest: PullRequest): Promise<void> {
+    const {prNumber, targetBranches, requiredBaseSha, needsCommitMessageFixup} = pullRequest;
     // In case a required base is specified for this pull request, check if the pull
```

Comment:
We seem to do much more than checking here, that seems unfortunate. Also the merge function below depends on the side effects from check before. Should we just check consistently outside of strategies similar to how we do it in the API merge?

Reply:
I extracted this check concept into the base merge strategy class, since checking if we can merge a pull request shouldn't actually be based on the type of merge.

Instead the merge strategies individually can just perform the actual merge.